drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c

```diff
@@ -0,0 +1,1272 @@
+		tx_buf = cmd->tx_buf->buf;
+		tx_len = cmd->tx_buf->len;
+	}
+
+	if (cmd->rx_buf) {
+		rx_buf = cmd->rx_buf->buf;
+		rx_len = cmd->rx_buf->len;
+	}
+
+	if ((rx_len != 0) && (tx_len != 0)) {
+		if (rx_len != tx_len)
+			return -EINVAL;
+
+		xfer_len += tx_len;
+	} else
+		/* At least one of these is zero. */
+		xfer_len += tx_len + rx_len;
+
+	if (xfer_len > NRF_QSPI_CINSTR_LEN_9B) {
+		LOG_WRN("cinstr %02x transfer too long: %zu", cmd->op_code, xfer_len);
+
+		return -EINVAL;
+	}
+
+	nrf_qspi_cinstr_conf_t cinstr_cfg = {
+		.opcode = cmd->op_code,
+		.length = xfer_len,
+		.io2_level = true,
+		.io3_level = true,
+		.wipwait = false,
+		.wren = wren,
+	};
+
+	qspi_lock(dev);
+
+	int res = nrfx_qspi_cinstr_xfer(&cinstr_cfg, tx_buf, rx_buf);
+
+	qspi_unlock(dev);
+	return qspi_get_zephyr_ret_code(res);
+}
+
+/* RDSR wrapper.  Negative value is error. */
+static int qspi_rdsr(const struct device *dev)
+{
+	uint8_t sr = -1;
+	const struct qspi_buf sr_buf = {
+		.buf = &sr,
+		.len = sizeof(sr),
+	};
+	struct qspi_cmd cmd = {
+		.op_code = SPI_NOR_CMD_RDSR,
+		.rx_buf = &sr_buf,
+	};
+	int ret = qspi_send_cmd(dev, &cmd, false);
+
+	return (ret < 0) ? ret : sr;
+}
+
+/* Wait until RDSR confirms write is not in progress. */
+static int qspi_wait_while_writing(const struct device *dev)
+{
+	int ret;
+
+	do {
+		ret = qspi_rdsr(dev);
+	} while ((ret >= 0) && ((ret & SPI_NOR_WIP_BIT) != 0U));
+
+	return (ret < 0) ? ret : 0;
+}
+
+/**
+ * @brief Fills init struct
+ *
+ * @param config Pointer to the config struct provided by user
+ * @param initstruct Pointer to the configuration struct
+ * @retval None
+ */
+static inline void qspi_fill_init_struct(nrfx_qspi_config_t *initstruct)
+{
+	bool quad_mode;
+
+	/* Configure XIP offset */
+	initstruct->xip_offset = 0;
+
+#ifdef CONFIG_PINCTRL
+	initstruct->skip_gpio_cfg = true,
+	initstruct->skip_psel_cfg = true,
+#else
+	/* Configure pins */
+	initstruct->pins.sck_pin = DT_PROP(QSPI_NODE, sck_pin);
+	initstruct->pins.csn_pin = QSPI_PROP_AT(csn_pins, 0);
+	initstruct->pins.io0_pin = QSPI_PROP_AT(io_pins, 0);
+	initstruct->pins.io1_pin = QSPI_PROP_AT(io_pins, 1);
+#if QSPI_PROP_LEN(io_pins) > 2
+	initstruct->pins.io2_pin = QSPI_PROP_AT(io_pins, 2);
+	initstruct->pins.io3_pin = QSPI_PROP_AT(io_pins, 3);
+#else
+	initstruct->pins.io2_pin = NRF_QSPI_PIN_NOT_CONNECTED;
+	initstruct->pins.io3_pin = NRF_QSPI_PIN_NOT_CONNECTED;
+#endif
+#endif /* CONFIG_PINCTRL */
+	/* Configure Protocol interface */
+#if DT_INST_NODE_HAS_PROP(0, readoc)
+	initstruct->prot_if.readoc =
+		(nrf_qspi_readoc_t)qspi_get_lines_read(DT_ENUM_IDX(DT_DRV_INST(0), readoc));
+#else
+	initstruct->prot_if.readoc = NRF_QSPI_READOC_FASTREAD;
+#endif
+
+#if DT_INST_NODE_HAS_PROP(0, writeoc)
+	initstruct->prot_if.writeoc =
+		(nrf_qspi_writeoc_t)qspi_get_lines_write(DT_ENUM_IDX(DT_DRV_INST(0), writeoc));
+#else
+	initstruct->prot_if.writeoc = NRF_QSPI_WRITEOC_PP;
+#endif
+	initstruct->prot_if.addrmode = _qspi_get_address_size(NRF_QSPI_ADDRMODE_24BIT);
```

Comment:
```suggestion
	initstruct->prot_if.addrmode = NRF_QSPI_ADDRMODE_24BIT;
```
You can also remove `qspi_get_address_size`, `_qspi_get_address_size`, `_qspi_get_readoc`, and `_qspi_get_writeoc`.